samples/caf/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(NONE)
+
+# NORDIC SDK APP START
+target_sources(app PRIVATE
+  src/main.c
+)
+# NORDIC SDK APP END
+
+# Add include directory for board specific CAF def files
+zephyr_include_directories(
+  configuration/${BOARD}
+  )
+
+target_sources_ifdef(CONFIG_CAF_LEDS
```

Comment:
Maybe you could introduce very simple Kconfig file to introduce an option to enable this module and show how to properly handle dependencies related to CAF events.

Reply:
I wanted to keep ingerention to Kconfig structure to minimum. Is it worth adding like this?

Reply:
I though that through. I'll do it as you suggested.